src/main/java/avd/inf/jdm/rentmycar/controller/RideController.java

```diff
@@ -0,0 +1,67 @@
+package avd.inf.jdm.rentmycar.controller;
+
+import avd.inf.jdm.rentmycar.domain.Offer;
+import avd.inf.jdm.rentmycar.domain.Ride;
+import avd.inf.jdm.rentmycar.service.RideService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api")
+public class RideController {
+
+private final RideService rideService;
+
+@Autowired
+    public RideController(RideService rideService) { this.rideService = rideService;}
+
+    @GetMapping("/v1/rides")
+    public ResponseEntity<List<Ride>> getAllRides() {
+    List<Ride> found = rideService.getAllRides();
+    return found.isEmpty() ?  ResponseEntity.noContent().build() : ResponseEntity.ok(found);
+    }
+    @GetMapping("/v1/rides/car/{carId}")
+    public ResponseEntity<List<Ride>> getRidesById(@PathVariable Long carId) {
+        try {
+            List<Ride> found = new ArrayList<>();
+            found.addAll(rideService.getRidesByCarId(carId));
+            return found.isEmpty() ? ResponseEntity.noContent().build() : ResponseEntity.ok(found);
+        } catch (Exception e) {
+            return new ResponseEntity<>(null, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+    }
+@PostMapping
```

Comment:
I am missing the endpoint here. As it is now, the endpoint for this POST request is /api right?
However, is the `PostMapping` even still needed? Since we create a Ride from the Booking?